nucypher/network/server.py

```diff
@@ -291,29 +291,37 @@ def revoke_arrangement(id_as_hex):
 
     @rest_app.route('/kFrag/<id_as_hex>/reencrypt', methods=["POST"])
     def reencrypt_via_rest(id_as_hex):
-        from nucypher.policy.collections import WorkOrder  # Avoid circular import
-        arrangement_id = binascii.unhexlify(id_as_hex)
+
+        # Get Policy Arrangement
+        try:
+            arrangement_id = binascii.unhexlify(id_as_hex)
+        except (binascii.Error, TypeError):
+            return Response(response=b'Invalid arrangement ID', status=405)
         try:
             with ThreadedSession(db_engine) as session:
                 arrangement = datastore.get_policy_arrangement(arrangement_id=id_as_hex.encode(), session=session)
         except NotFound:
             return Response(response=arrangement_id, status=404)
-        kfrag_bytes = policy_arrangement.kfrag  # Careful!  :-)
-        verifying_key_bytes = policy_arrangement.alice_verifying_key.key_data
 
-        # TODO: Push this to a lower level. Perhaps to Ursula character? #619
-        kfrag = KFrag.from_bytes(kfrag_bytes)
-        alices_verifying_key = UmbralPublicKey.from_bytes(verifying_key_bytes)
-        alices_address = canonical_address_from_umbral_key(alices_verifying_key)
+        # Get KFrag
+        kfrag = KFrag.from_bytes(arrangement.kfrag)  # Careful!  :-)
```

Comment:
I think I asked about the `# Careful!  :-)` bit like a year ago and no one knows why we should be careful. Let's remove it.

Reply:
I was concerned that `kFrag` is optional to `Arrangement`, I still agree that it can be removed. :heavy_check_mark: